woocommerce/woocommerce-android
# [Better Errors] Handle timeout errors inside Orders list

Summary
==========
Fix issue #10796 by introducing a Retry event whenever a Timeout error is detected during the Orders initial loading.

This PR depends on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2960. Please make sure to review that PR before merging this one.

How to Test
==========
### Test setup
First, install the MacOS `Network Link Conditioner` software from the [Xcode Additional Tools](https://developer.apple.com/download/all/). This software, once installed, will be available in your Mac settings and allow you to configure network limiters that will make it easier to reproduce Timeout scenarios inside an emulator.

I suggest using a custom network configuration with a very restricted network. This way, we can ensure that the device still detects a connection, but it takes forever to finish any request.

<img width="878" alt="Screenshot 2024-02-28 at 23 12 14" src="https://github.com/woocommerce/woocommerce-android/assets/5920403/dbf24c2b-4917-41b4-b71e-504cb3dae77e">
<img width="676" alt="Screenshot 2024-02-28 at 23 12 57" src="https://github.com/woocommerce/woocommerce-android/assets/5920403/5d646c0d-24b9-40c7-b257-39bd1864133e">

Feel free to reach out if any help is needed with this setup.

### Test scenario
⚠️ If you want to check when the Retry happens visually, add a Log to the `OrderListViewModel` in line 421.

1. Open the app, go to the Orders section
2. Once inside the Orders list, wait for the Timeout error to happen first
3. Verify the loading Retries

Update release notes:
